conans/client/command.py

```diff
@@ -1332,12 +1354,44 @@ def alias(self, *args):
 
         self._conan.export_alias(args.reference, args.target)
 
+    def link(self, *args):
+        """ Links a conan reference (e.g lib/1.0@conan/stable) with a local folder path.
+
+        """
+        parser = argparse.ArgumentParser(description=self.link.__doc__,
+                                         prog="conan link")
+        parser.add_argument('target', help='Path to the package folder in the user workspace',
+                            nargs='?', action=ConanFilePathOptional)
+        parser.add_argument('reference', help='Reference to link. e.g.: mylib/1.X@user/channel')
+        parser.add_argument("--remove", action='store_true', default=False,
+                            help='Remove linked reference (target not required)')
+
+        args = parser.parse_args(*args)
+        self._warn_python2()
+
+        # Args sanity check
+        if args.remove and args.target:
+            raise ConanException("Do not provide the 'target' argument for removal")
+
+        if not args.remove and not args.target:
+            raise ConanException("Argument 'target' is required to link a reference")
+
+        reference = ConanFileReference.loads(args.reference)
```

Comment:
Removing this ConanFileReference from here, it will be created inside the _conan_api_